docs/developer-guide.rst

```diff
@@ -0,0 +1,629 @@
+.. set temporal working directory
+.. jupyter-execute::
+    :hide-code:
+
+    import os
+    import audeer
+
+    _cwd_root = os.getcwd()
+    _tmp_root = audeer.mkdir(os.path.join('docs', 'tmp-developer-guide'))
+    os.chdir(_tmp_root)
+
+
+.. _developer-guide:
+
+Developer guide
+===============
+
+The aim of
+:mod:`audbackend`
+is to provide an
+abstract interface for
+any kind of file storage system.
+Even those,
+that have not been
+invented yet :)
+
+This tutorial is divided
+into three parts.
+Under :ref:`register-a-backend`,
+we show how existing or new backends
+are assigned to names,
+that are used by
+:func:`audbackend.create`
+and
+:func:`audbackend.access`.
+In :ref:`develop-new-interface`,
+we show how to create an interface
+that manages user content.
+Under :ref:`develop-new-backend`,
+we take a deep dive
+and develop a backend
+that stores files into
+a SQLite_ database.
+
+
+.. _register-a-backend:
+
+Register a backend
+------------------
+
+Backends are referred to by names.
+The current active names can be listed by:
+
+.. jupyter-execute::
+
+    import audbackend
+
+    list(audbackend.available())
+
+We can assign a name to a backend
+by registering its backend class.
```

Comment:
```suggestion
We register a backend class
by assigning an alias to it.
Functions like
:func:`audbackend.create`
and
:func:`audbackend.access`
expect the alias instead
of the class name.
This makes it easier
to read available repositories
from a file.
```

Reply:
The last sentence I don't get. You mean it is easier to define repositories in config files, correct?

Reply:
Yes, I think that was actually the motivation why we introduced that whole concept of alias names. In the end, we need to keep track of the repositories we e.g. want to search for databases in case of audb.

Reply:
I applied your suggested changes, and added `(config)` in front of `file`.